### ISSUE TITLE
org.apache.tools.ant.taskdefs.optional.ANTLR was not found when building

### ISSUE THREAD

**Issue by diogoleal** [processing/processing#3502]
I'm compiling version 3.0a11, but is experiencing an error when I run "ant run."
```
BUILD FAILED
/home/diogo/Downloads/processing-processing-0238-3.0a11/build/build.xml:286: The following error occurred while executing this line:
/home/diogo/Downloads/processing-processing-0238-3.0a11/build/build.xml:336: The following error occurred while executing this line:
/home/diogo/Downloads/processing-processing-0238-3.0a11/java/build.xml:37: Problem: failed to create task or type antlr
Cause: the class org.apache.tools.ant.taskdefs.optional.ANTLR was not found.
        This looks like one of Ant's optional components.
Action: Check the optional que JAR Appropriate exists in
        - / Usr / share / apache-ant / lib
        - / Home / diogo / .ant / lib
        -a directory added on the command line with the argument -LIB
 
Do not panic, this is a common problem.
The commonest cause is a missing JAR.
 
This is not a bug; it is a configuration problem
```

The full error message is here -> http://pastebin.com/rQ9ZQXzv

From the little I have understanding of the message, it seems to lack the antlr, did his installation, yet the error message is displayed.

I did the tests running antlr via command line and appeared to function properly.
```
Downloads % antlr   
ANTLR Parser Generator  Version 4.5
 -o ___              specify output directory where all output is generated
 -lib ___            specify location of grammars, tokens files
```
Yet the error message does not change.

I downloaded ```antlr-4.5-complete.jar``` and ```antlr-runtime-4.5.1.jar``` and copy to the ```/usr/share/apache-ant/lib/``` rename these files by removing the version(ex. antlr-runtime.jar), and also tried to leave only ```antlr.jar```, yet the
 compilation does not work.

The reason I'm compiling this is what I'm packing the processing for the [Void Linux](http://voidlinux.eu)

**Comment by benfry** [comment]
That looks like you have an alternate version of antlr somewhere in your CLASSPATH. The build instructions are correct as they are: there's no need to install antlr; you need the exact version that's included in the repo. Other versions are not compatible, which is what you're seeing here.

**Comment by benfry** [comment]
No response; closing.

**Comment by spthaolt** [comment]
Hello,

I have just encounter the same issue when trying to compile processing from source (branch master ,cloned on June 19 2016). 

My system specs:
Debian 8 x86_64
java-8-oracle 
ant-1.9.7

At first I tried to manipulate env. CLASSPATH variable. No success. Then tried to edit build/shared/tools/MovieMaker/build.xml (task javac) to include path to $ANT_HOME/lib, because I thought it's need ant-antlr.jar, still no luck.

Finally, I downloaded ant-optional-1.5.4.jar from http://mvnrepository.com/artifact/ant/optional/1.5.4 and put it in $ANT_HOME/lib. Everything works. Processing was built and run. Hope this would help anyone having the same issue.
